### PR TITLE
Jumble text

### PR DESCRIPTION
**Pull request by mariuszdb**
I'm creating this PR to implement a text shuffler https://github.com/tridactyl/tridactyl/issues/2733 (I called it "jumble").
I decided to treat a word as a sequence of one of: only lowercase letters or only uppercase letters or an uppercase letter followed by only lowercase letters (word or WORD or Word). For example, a variable in code may be named "someVariableName" and it makes little sense to treat it like one word so it's treated like 3 words ("some", "Variable" and "Name") and the letters are shuffled separately in each one. The same goes for company names and other things written in camelCase/PascalCase. Characters that are not letters are never shuffled.
I do this by splitting the text using this regex: ([^a-zA-Z]|[A-Z][a-z]+).
I based the solution on rot13_helper.
I'm attaching a few screenshots of websites with the text shuffled.
image
image
image

Just for fun.

### CODE REVIEW

**Comment by bovine3dom** [comment]
If we follow the camel case logic I don't think acronyms should be shuffled. A decent heuristic could be: a word in all caps of four letters or fewer is an acronym.

File: src/excmds.ts [hunk 1]
```diff
@@ -4363,6 +4363,22 @@ export function rot13(n: number) {
         body.currentNode.textContent = rot13_helper(t, n)
     }
 }
+/**
+ * Perform text jumbling.
+ *
+ * Shuffles letters except for first and last in all words in text nodes in the current tab. Only characters in
+ * the ASCII range are considered.
```
Comment (bovine3dom) [inline src/excmds.ts:4370]: Should probably add a reference to a site that explains the phenomenon : )

File: src/lib/config.ts [hunk 2]
```diff
@@ -239,6 +239,7 @@ export class default_config {
         x: "stop",
         gi: "focusinput -l",
         "g?": "rot13",
+        "g!": "jumble",
```
Comment (bovine3dom) [inline src/lib/config.ts:242]: Can we add an ex alias for `reibadailty`? I liked that joke

File: src/excmds.ts (hunk 1 above)
Comment (mariuszdb) [reply src/excmds.ts:4370]: Added

File: src/lib/config.ts (hunk 2 above)
Comment (mariuszdb) [reply src/lib/config.ts:242]: Added

**Comment by mariuszdb** [comment]
> If we follow the camel case logic I don't think acronyms should be shuffled. A decent heuristic could be: a word in all caps of four letters or fewer is an acronym.

Added, I was thinking of skipping uppercase words altogether but I guess this is ok.

**Comment by bovine3dom** [comment]
This is my new favourite thing, thank you

**Pull request merged**